sw/device/lib/dif/dif_entropy.c

```diff
@@ -93,12 +93,40 @@ dif_entropy_result_t dif_entropy_configure(const dif_entropy_t *entropy,
   return kDifEntropyOk;
 }
 
+inline static bool get_es_valid(const dif_entropy_t *entropy) {
+  uint32_t reg = mmio_region_read32(entropy->params.base_addr,
+                                    ENTROPY_SRC_INTR_STATE_REG_OFFSET);
+  return bitfield_bit32_read(reg, ENTROPY_SRC_INTR_STATE_ES_ENTROPY_VALID_BIT);
+}
+
+inline static void clr_es_valid(const dif_entropy_t *entropy) {
+  mmio_region_write32(entropy->params.base_addr,
+                      ENTROPY_SRC_INTR_STATE_REG_OFFSET,
+                      1 << ENTROPY_SRC_INTR_STATE_ES_ENTROPY_VALID_BIT);
+}
+
 dif_entropy_result_t dif_entropy_read(const dif_entropy_t *entropy,
                                       uint32_t *word) {
   if (entropy == NULL || word == NULL) {
     return kDifEntropyBadArg;
   }
+
+  // first wait for entropy data to be available
+  while (!get_es_valid(entropy))
```

Comment:
We can do the following to let the caller figure out how to proceed with the polling:

```c
if (!get_es_valid(entropy)) {
  return kDifEntropyDataUnavailable;
}
```

Reply:
yep fixed! please take a look.